app/src/main/java/mihon/feature/upcoming/UpcomingScreenContent.kt

```diff
@@ -140,7 +175,13 @@ private fun UpcomingScreenSmallImpl(
                     )
                 }
                 is UpcomingUIModel.Header -> {
-                    ListGroupHeader(text = relativeDateText(item.date))
+                    val mangaCount = items.filterIsInstance<UpcomingUIModel.Item>()
+                        .count { it.manga.expectedNextUpdate?.toLocalDate() == item.date }
+
+                    DateHeading(
+                        date = item.date,
+                        mangaCount = mangaCount,
+                    )
```

Comment:
This calculation should be done inside the screen model. If we do this in the UI the calculation will need to run again after the UI element goes off screen (plus some more).

Reply:
IDK if what I did is what you meant but we will see.